utils/errors.go

```diff
@@ -22,3 +26,30 @@ func TypeStr[T any]() string {
 	vT := reflect.TypeOf(zero).Elem()
 	return vT.String()
 }
+
+// NewBuildTimeoutError is used when a resource times out during construction or reconfiguration.
+func NewBuildTimeoutError(name string) error {
+	timeout := GetResourceConfigurationTimeout(logging.Global())
+	id := fmt.Sprintf("resource %s", name)
+	timeoutMsg := "reconfigure"
+	return timeoutErrorHelper(id, timeout, timeoutMsg, DefaultResourceConfigurationTimeout, ResourceConfigurationTimeoutEnvVar)
+}
+
+// NewModuleStartUpTimeoutError is used when a module times out during startup.
+func NewModuleStartUpTimeoutError(name string) error {
+	timeout := GetModuleStartupTimeout(logging.Global())
+	id := fmt.Sprintf("module %s", name)
+	timeoutMsg := "startup"
+	return timeoutErrorHelper(id, timeout, timeoutMsg, DefaultModuleStartupTimeout, ModuleStartupTimeoutEnvVar)
+}
+
+func timeoutErrorHelper(id string, timeout time.Duration, timeoutMsg string, defaultTimeout time.Duration, timeoutEnvVar string) error {
+	extraMsg := ""
+	if timeout == defaultTimeout {
+		extraMsg = fmt.Sprintf(" Update %s env variable to override", timeoutEnvVar)
+	}
+	return fmt.Errorf(
+		"%s timed out after %v during %v.%v",
```

Comment:
Is this the same error message as before or a different one? I can't quite tell with all the formatting directives.

Reply:
Yeah that seems better; I was worried people may be relying on the language of the original log message but I'm probably being overly defensive.